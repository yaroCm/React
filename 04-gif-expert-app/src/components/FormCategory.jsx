import { useState } from 'react';

export const FormCategory = ({ addCategory }) => {
  const [value, setValue] = useState('');

  const onChangeInputSearch = ({ target }) => {
    setValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    if (value.trim().length <= 1) return;

    addCategory(value.trim());

    setValue('');
  };
  return (
    <form name='form' onSubmit={onSubmit}>
      <input
        type='search'
        onChange={onChangeInputSearch}
        value={value}
        placeholder='Search gifts'
      />
      <button type='submit'>Add</button>
    </form>
  );
};
