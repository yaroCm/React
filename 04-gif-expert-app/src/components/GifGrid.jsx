import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando</h2>}
      <div className='card-grid'>
        {gifs.map((gif) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
