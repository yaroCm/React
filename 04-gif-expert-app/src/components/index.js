export * from './FormCategory';
export * from './GifGrid';
export * from './GifItem';
