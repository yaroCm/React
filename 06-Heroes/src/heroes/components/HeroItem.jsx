import { Link } from 'react-router-dom';

const CharacterByHero = ({ characters, alter_ego }) =>
  characters === alter_ego ? (
    <p className='card-text'>{alter_ego}</p>
  ) : (
    <p className='card-text'>{characters}</p>
  );

export const HeroItem = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {
  const heroImg = `/assets/heroes/${id}.jpg`;
  return (
    <div className='col '>
      <div className='card'>
        <div className='row no-gutters'>
          <div className='col-4'>
            <img src={heroImg} className='card-img' alt={superhero} />
          </div>
          <div className='col-8'>
            <div className='card-body'>
              <h5 className='card-title'>{superhero}</h5>
              <CharacterByHero characters={characters} alter_ego={alter_ego} />
              <p className='card-text'>{first_appearance}</p>
              <Link to={`/hero/${id}`}>Mas...</Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
