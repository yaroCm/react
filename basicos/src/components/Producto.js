import React from 'react'

const Producto = ({productos,producto,carrito,setCarrito}) => {
    const {nombre,id,precio}=producto;

    const addProduct=id=>{
        const product = productos.filter(producto=>producto.id===id)[0];
        setCarrito([...carrito,product]);
    }
    const deleteProduct=id=>{
        const productos =carrito.filter(producto=>producto.id!=id);
        setCarrito(productos);
    }
    return (  
       <div>
           <h2>{nombre}</h2>
           <p>${precio}</p>
           {productos?(
               <button 
               type="button"
               onClick={()=>addProduct(id)}>
                   Comprar
               </button>
           )
           :(
            <button 
            type="button"
            onClick={()=>deleteProduct(id)}>
                Eliminar
            </button> 
           )}
           
       </div>
    );
}
 
export default Producto;