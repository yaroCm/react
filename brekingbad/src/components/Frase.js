import styled from '@emotion/styled';
import React from 'react';
const DivFrase=styled.div`
    padding: 3rem;
    border-radius: .5rem;
    background-color: #fff;
    max-width: 800px;
    margin-top: 10rem;

    h1{
        font-family: Arial, Helvetica, sans-serif;
        text-align: center;
        position: relative;
        padding-left: 2rem;

        &::before{
            content: open-quote;
            color: black;
            position: absolute;
            left: -1rem;
            top: -2rem;
        }
    }
    p{
        font-family: Verdana, Geneva, Tahoma, sans-serif ;
        font-weight:bold;
        font-size: 1.6rem;
        text-align: right;
        color: #666;
        margin-top: 2rem;
    }
`;
const Frase = ({frase}) => {
    return (
        <DivFrase>
            <h1>{frase.quote}</h1>
            <p>{frase.author}</p>
        </DivFrase>
    );
}
 
export default Frase;