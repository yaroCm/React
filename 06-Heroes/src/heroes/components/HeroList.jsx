import React, { useMemo } from 'react';
import { GetHeroByPublisher } from '../helpers/GetHeroByPublisher';
import { HeroItem } from './HeroItem';

export const HeroList = ({ publisher }) => {
  const heros = useMemo(() => GetHeroByPublisher(publisher), [publisher]);
  return (
    <div className='row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn'>
      {heros.map((hero) => (
        <HeroItem key={hero.id} {...hero} />
      ))}
    </div>
  );
};
