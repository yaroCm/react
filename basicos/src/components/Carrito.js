import React from 'react';
import './carrito.css';
import Producto from './Producto';
const Carrito = ({carrito,setCarrito}) => (
    <div className="carrito">
        <h2>Carrito de compras</h2>
        {carrito.length===0?
        <p>No hay elementos en el carrito</p>
        :
        carrito.map(producto=>(
            <Producto
                key={producto.id}
                producto={producto}
                carrito={carrito}
                setCarrito={setCarrito}
            />
        ))}
    </div>
);
    

 
export default Carrito;