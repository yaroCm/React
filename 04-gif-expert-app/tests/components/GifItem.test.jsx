/**
 * @jest-environment jsdom
 */
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas con <GifItem/>', () => {
  const title = 'jordi';
  const url = 'http://localhost/odermygear.com';

  test('match snapsnot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Show URL and Title <img>', () => {
    render(<GifItem title={title} url={url} />);

    //opcion simple
    expect(screen.getByRole('img').src).toBe(url);
    expect(screen.getByRole('img').alt).toBe(title);

    //opcion elegante
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Show the title ', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
