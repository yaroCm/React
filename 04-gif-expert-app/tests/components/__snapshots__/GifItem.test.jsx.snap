// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifItem/> match snapsnot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="jordi"
      src="http://localhost/odermygear.com"
    />
    <p>
      jordi
    </p>
  </div>
</div>
`;
